numpy/lib/tests/test_format.py

```diff
@@ -963,3 +963,34 @@ def test_unicode_field_names():
     with open(fname, 'wb') as f:
         with assert_warns(UserWarning):
             format.write_array(f, arr, version=None)
+
+
+@pytest.mark.parametrize('dt, fail', [
+    (np.dtype({'names': ['a', 'b'], 'formats':  [float, np.dtype('S3',
+                 metadata={'some': 'stuff'})]}), True),
+    (np.dtype(int, metadata={'some': 'stuff'}), False),
+    (np.dtype(int, (2,), metadata={'some': 'stuff'}), False),
+    # recursive: metadata on the field of a dtype
+    (np.dtype({'names': ['a', 'b'], 'formats': [float,
+        np.dtype({'names': ['c'], 'formats': [np.dtype(int, metadata={})]})],
+        }),
+     False)
```

Comment:
Nit: I'd find this wrapping easier to follow as
```
(np.dtype({'names': ['a', 'b'], 'formats': [
    float, np.dtype({'names': ['c'], 'formats': [np.dtype(int, metadata={})]})
]}), False)
```

Reply:
ok